### ISSUE TITLE
Problems with module format when using "rollup"

### ISSUE THREAD

**Issue by TomStrepsil** [wuifdesign/postcss-replace-values#1]
Thanks so much for this plugin!

It's working great for me in a webpack setup using [postcss-loader](https://github.com/webpack-contrib/postcss-loader) etc.

For some reason when using with rollup, the [rollup-plugin-postcss ](https://www.npmjs.com/package/rollup-plugin-postcss) can't parse the module format.  And I'm having no joy trying to configure [@rollup/plugin-commonjs](https://www.npmjs.com/package/@rollup/plugin-commonjs) to perhaps handle this (if that would be a solution...) without interfering with [@rollup/plugin-babel](https://www.npmjs.com/package/@rollup/plugin-babel).

The error I get is:
```
[!] (plugin postcss) Error: Loading PostCSS Plugin failed: req(...) is not a function
```
...Seemingly like it can't understand the export syntax, which appears to be compiled for interop:
```
exports.__esModule = true;
exports["default"] = postcss.plugin('replace-values', function (options) {
```
...changing this to commonjs format:
```
module.exports = postcss.plugin('replace-values', function (options) {
```
...makes everything work.

Would you consider other module formats for this package, or consider a pull request to modify it?

**Comment by wuifdesign** [comment]
it seams like this is caused by the build with "tsc". but i'll take a look if i can change the build to use `module.exports`.

**Comment by wuifdesign** [comment]
for me it works fine with rollup, do you have an example where it throws this error. i used `"rollup": "^2.60.0",` in my demo. it worked fine:

```
  "dependencies": {
    "react": "^17.0.1",
    "react-dom": "^17.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.16.0",
    "@babel/preset-react": "^7.16.0",
    "@rollup/plugin-babel": "^5.3.0",
    "@rollup/plugin-commonjs": "^21.0.1",
    "@rollup/plugin-image": "^2.1.1",
    "@rollup/plugin-node-resolve": "^13.0.6",
    "@rollup/plugin-replace": "^3.0.0",
    "postcss-replace-values": "^0.0.1",
    "rollup": "^2.60.0",
    "rollup-plugin-livereload": "^2.0.5",
    "rollup-plugin-postcss": "^4.0.1",
    "rollup-plugin-serve": "^1.1.0"
  }
```

```
import serve from 'rollup-plugin-serve';
import livereload from 'rollup-plugin-livereload';
import babel from '@rollup/plugin-babel';
import { nodeResolve } from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import replace from '@rollup/plugin-replace';
import image from '@rollup/plugin-image';
import postcss from 'rollup-plugin-postcss';

const replaceValuesPlugin = require('postcss-replace-values').default;

export default {
  input: 'src/index.js',
  output: {
    file: 'dist/bundle.js',
    format: 'iife',
    sourcemap: true,
  },
  plugins: [
    image(),
    postcss({
      extensions: ['.css'],
      plugins: [
        replaceValuesPlugin({
          values: {
            '#000': '#a00',
          },
        }),
      ],
    }),
    nodeResolve({
      extensions: ['.js'],
    }),
    replace({
      'process.env.NODE_ENV': JSON.stringify('development'),
    }),
    babel({
      presets: ['@babel/preset-react'],
    }),
    commonjs(),
    serve({
      open: true,
      verbose: true,
      contentBase: ['', 'public'],
      host: 'localhost',
      port: 3000,
    }),
    livereload({ watch: 'dist' }),
  ],
};
```

**Comment by TomStrepsil** [comment]
Thanks for the speedy response! I should've been clearer; it does work fine with the "[array format](https://github.com/postcss/postcss-load-config#array)" for `postcss.config.js` (since can access the default export and new up the plugin), but doesn't seem to work with the "[object format](https://github.com/postcss/postcss-load-config#object)" (where plugins aren't instantiated, but passed as `[plugin-name]: {}`) - it's probably some vagary with my babel setup, but finding it hard to find a workaround.

**Comment by TomStrepsil** [comment]
To illustrate; taking your example, and changing the rollup config to use `postcss` without in-line configuration:
```
import serve from 'rollup-plugin-serve';
import livereload from 'rollup-plugin-livereload';
import babel from '@rollup/plugin-babel';
import { nodeResolve } from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import replace from '@rollup/plugin-replace';
import image from '@rollup/plugin-image';
import postcss from 'rollup-plugin-postcss';

export default {
  input: 'src/index.js',
  output: {
    file: 'dist/bundle.js',
    format: 'iife',
    sourcemap: true,
  },
  plugins: [
    image(),
    postcss(),
    nodeResolve({
      extensions: ['.js'],
    }),
    replace({
      'process.env.NODE_ENV': JSON.stringify('development'),
    }),
    babel({
      presets: ['@babel/preset-react'],
    }),
    commonjs(),
    serve({
      open: true,
      verbose: true,
      contentBase: ['', 'public'],
      host: 'localhost',
      port: 3000,
    }),
    livereload({ watch: 'dist' }),
  ],
};
```
...then creating a `postcss.config.js` (using the aforementioned ["object format"](https://github.com/postcss/postcss-load-config#object)):
```
module.exports = () => ({
  "plugins": {
    "postcss-replace-values": {
      "values": {
        "#000": "#a00"
      }
    }
  }
});
```
...produces the error:
```
[!] (plugin postcss) Error: Loading PostCSS Plugin failed: req(...) is not a function
```
So it seems the module format used is incompatible with defining your [postcss config in package.json](https://github.com/postcss/postcss-load-config#packagejson) or a [config](https://github.com/postcss/postcss-load-config#postcssrc) [file](https://github.com/postcss/postcss-load-config#postcssrcjs-or-postcssconfigjs).

I cannot explain why I don't get the same issue when using `postcss-loader` etc with webpack with the equivalent configuration file.  I've used identical babel setups either way, and it only seems to fail when using this rollup setup.

I appreciate there _is_ a way to use the plugin by requiring `{ default: plugin } = require("postcss-replace-values")` then pasing `plugin` to the [postcss api](https://postcss.org/api/) or by using the [array format](https://github.com/postcss/postcss-load-config#array) in a config, but would be great to support all the configuration styles postcss supports/recommends, since all the other plugins I've come across seem to work in this manner.

If you have any tweaks to the typescript compiler that you think could work for all cases, would really appreciate it!

Thanks again

**Comment by TomStrepsil** [comment]
Sadly it looks like TypeScript is doing the right thing here, but support is flaky for the output.

[Babel 7.14.0](https://babeljs.io/blog/2021/04/29/7.14.0) added an `importInterop: "node"` feature to better match nodes module resolution algorithms.  This apparently was also added to `@rollup/plugin-commonjs`:
> Nicolò from our team also contributed a similar option to @rollup/plugin-commonjs, which will be out in the next release. Our goal is to help the ecosystem migrate to native ECMAScript modules by providing an easier migration path.

...as [defaultIsModuleExports](https://github.com/rollup/plugins/tree/master/packages/commonjs/#defaultIsModuleExports), but sadly all experiments with that (and [esmExternals](https://github.com/rollup/plugins/tree/master/packages/commonjs/#esmexternals)) don't seem to work, for me.

Looks like this is just a mismatch between rollup's commonjs support and typescript compiler...

**Comment by wuifdesign** [comment]
hm, i see. but i can't find a way to make `tsc` to export the `module.exports` syntax. do you see a way?  I just wanted to keep this lib simple so i just used `tsc`. you may send a pull request if you find a solution?

```
"build": "tsc lib/index.ts --declaration --outDir dist --skipLibCheck",
```

**Comment by TomStrepsil** [comment]
Sadly, I couldn't find a way to make `tsc` output commonjs format, but thanks for trying!  If I get any success and you'd happily receive a pull request, I will raise one.   Thanks again!

**Comment by wuifdesign** [comment]
should be fixed in `v1.0.0`.
